src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -695,6 +707,17 @@ public override void OnHeader(int index, bool indexOnly, ReadOnlySpan<byte> name
         }
     }
 
+    // Final: Has the message been fully received? We allow up to 2x grace while receiving the message
+    // to avoid faulting the entire connection. We check again later when we can reject the request with a 431.
+    protected override void CheckRequestHeadersCountLimit(bool final = false)
+    {
+        if (final && RequestHeadersParsed > ServerOptions.Limits.MaxRequestHeaderCount
+            || RequestHeadersParsed > ServerOptions.Limits.MaxRequestHeaderCount * 2)
```

Comment:
I think we can just get rid of this complicated condition, and I know that `||` has lower precedence than `&&`, but at least use parentheses for stuff like this 😆